pkg/controllers/networkpolicy/policy_controller.go

```diff
@@ -248,16 +209,44 @@ func (c *PolicyController) syncToCalico(key string) error {
 		return nil
 	} else {
 		// The object exists - update the datastore to reflect.
-		log.Infof("Add/Update Policy %s in Calico datastore", key)
-		p := obj.(api.Policy)
-		_, err := c.calicoClient.Policies().Apply(&p)
-		return err
+		log.Infof("Create/Update NetworkPolicy %s in Calico datastore", key)
+		p := obj.(api.NetworkPolicy)
+
+		// Lookup to see if this object already exists in the datastore.
+		gp, err := c.calicoClient.NetworkPolicies().Get(context.Background(), p.Namespace, p.Name, options.GetOptions{})
```

Comment:
for passing the contexts, we should usually create the context from where these controllers or this instance is getting called from, instead of individual `context.Background()` for every call. We don't utilize context to cancel any calls so not too fussed about it